src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -4606,7 +4606,19 @@ public String getDrainProgress()
      * - Drain waits for in-progress streaming to complete
      * - Drain flushes *all* columnfamilies (shutdown hook only flushes non-durable CFs)
      */
-    public synchronized void drain() throws IOException, InterruptedException, ExecutionException
+    public void drain() throws IOException, InterruptedException, ExecutionException
+    {
+        if (daemon.setupCompleted())
+        {
+            drainInternal();
+        }
+        else
+        {
+            throw new IllegalStateException("Cannot drain a node that is bootstrapping");
+        }
```

Comment:
this would happen if we called drain during regular startup as well, not just bootstrapping right

Reply:
then there's a question: if we're starting up vs bootstrapping is it desirable to wait for startup to complete then drain, or just fail the drain and continue w/ other shutdown actions later?

Reply:
The only case we should expect Drain to be called when the server is not fully initialized is if someone is taking manual action to terminate a node that is stuck initializing, or taking manual action to abort an in-progress bootstrap or node replacement. These are probably both cases we'd want to terminate without delay.